apps/test/unit/lib/levelbuilder/lesson-editor/ProgrammingExpressionsEditorTest.jsx

```diff
@@ -26,17 +48,30 @@ describe('ProgrammingExpressionsEditor', () => {
         }
       ],
       programmingEnvironments: [],
+      addProgrammingExpression,
       removeProgrammingExpression
     };
   });
 
+  afterEach(() => {
+    restoreRedux();
+  });
+
   it('renders default props', () => {
-    const wrapper = mount(<ProgrammingExpressionsEditor {...defaultProps} />);
+    const wrapper = mount(
+      <Provider store={store}>
```

Comment:
is this actually necessary, given that `ProgrammingExpressionsEditor` is actually `UnconnectedProgrammingExpressionsEditor`, as per line 5?

Reply:
the FindProgrammingExpressionDialog which is a sub component also needs the store and so it won't work without it

Reply:
is that still relevant if we switch to `shallow` rendering?

Reply:
If we switch to shallow rendering some of the other checks in the exists tests don't work such as looking for parts of the table

Reply:
right; hence my other comment about that. The point is that shallow rendering makes things simpler in every sense; it keeps tests focused on what they should be testing, which means both that we don't need to look inside other components to see what they're rendering and that we don't need to look inside other components to see what their rendering requirements are

Reply:
That makes sense. Since these were existing tests and this is something we would really like to have for Tuesday I'm going to keep it as is for right now. If you think we should track it to come back to we can do that.

Reply:
https://codedotorg.atlassian.net/browse/PLAT-894